bundle/Command/UpdateContentVisibilityCommand.php

```diff
@@ -0,0 +1,220 @@
+                        fn () => $this->repository->getContentLanguageService()->loadLanguageById($result['initial_language_id']),
+                    );
+                } catch (NotFoundException $exception) {
+                    $this->logger->error(
+                        sprintf(
+                            'Language with id #%d does not exist: %s',
+                            $languageId,
+                            $exception->getMessage(),
+                        ),
+                    );
+
+                    continue;
+                }
+
+                try {
+                    $contentId = $result['id'];
+
+                    /** @var Content $content */
+                    $content = $this->repository->sudo(
+                        fn () => $this->repository->getContentService()->loadContent($result['id'], [$language->getLanguageCode()]),
+                    );
+                } catch (NotFoundException $exception) {
+                    $this->logger->error(
+                        sprintf(
+                            'Content with id #%d does not exist: %s',
+                            $contentId,
+                            $exception->getMessage(),
+                        ),
+                    );
+
+                    continue;
+                }
+
+                try {
+                    $action = $this->scheduledVisibilityService->updateVisibilityIfNeeded($content);
+                } catch (InvalidStateException $exception) {
+                    $this->logger->error($exception->getMessage());
+
+                    continue;
+                }
+
+                if ($action !== VisibilityUpdateResult::NoChange) {
+                    $this->logger->info(
+                        sprintf(
+                            "Content '%s' with id #%d has been %s.",
+                            $content->getName(),
+                            $content->getId(),
+                            $action->value,
+                        ),
+                    );
+                }
+                $this->style->progressAdvance();
+            }
+            $offset += $limit;
+            $results = $pager->getAdapter()->getSlice($offset, $limit);
+        }
+        $this->style->progressFinish();
+
+        $this->style->info('Done.');
+
+        return Command::SUCCESS;
+    }
+
+    private function getQueryBuilder(): QueryBuilder
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->select('id', 'initial_language_id')
+            ->from('ezcontentobject');
+
+        return $query;
+    }
+
+    private function applyContentTypeLimit(QueryBuilder $query, array $contentTypeIds): QueryBuilder
+    {
+        $query = $query->where(
+            $query->expr()->in('contentclass_id', ':content_type_ids'),
+        )->setParameter('content_type_ids', $contentTypeIds, Connection::PARAM_INT_ARRAY);
+
+        return $query;
```

Comment:
Since objects are passed by reference, you don't need to return it here.

Reply:
fixed in 830d104618d4fbe897df495e9fcccbd9b4565543